sources/router.c

```diff
@@ -325,6 +328,19 @@ od_router_attach(od_router_t *router, od_config_t *config, od_client_t *client,
 	od_server_t *server;
 	for (;;)
 	{
+		if (!route->rule->db_state->is_active)
```

Comment:
Is there any timeout for routing?
Also, we should check wether client disconnected during wait.

Reply:
I will address this a bit later.

Reply:
I've studied postgess messaging protocol for any timeouts given by the client and haven't found anything. Where should we get routing timeout? Did I miss such a timeout in client's requests or should this timeout by configurable just in the pooler's configs?

Reply:
Talking about waiting I wander if the way I implemented it is correct. At the bottom of the same for cycle some other waiting is implemented by means of od_route_wait. But I'm not sure what and why it actually does.

Also I don't know how to determine if the client is still connected.